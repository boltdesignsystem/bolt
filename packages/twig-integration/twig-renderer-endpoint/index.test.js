const { render } = require('@bolt/twig-renderer');
const buttonTemplate = '@bolt-components-button/button.twig';

describe('Test the Bolt Twig Renderer API', () => {
  test('renders a basic button', async () => {
    const buttonData = {
      text: 'Hello world!',
    };
    const result = await render(buttonTemplate, buttonData);

    expect(result.ok).toEqual(true);
  });

  test('handles missing data', async () => {
    const result = await render(buttonTemplate);

    expect(result.ok).toEqual(true);
  });

  test('handles non-existent template path', async () => {
    const result = await render('@bolt-components-button/button2.twig', {
      text: 'Hello world 2!',
    });

    expect(result.ok).toEqual(false);
  });

  test('renders the button component correctly', async () => {
    const result = await render(buttonTemplate, {
      text: 'Hello world!2',
    });

    expect(result.html).toMatchSnapshot();
  });

  test('renders attributes on the button component correctly', async () => {
    const result = await render(buttonTemplate, {
      text: 'Hello world!',
      attributes: {
        class: ['u-bolt-margin-bottom-large'],
      },
    });

    expect(result.html).toMatchSnapshot();
  });

  test('renders the secondary button correctly', async () => {
    const result = await render(buttonTemplate, {
      text: 'Secondary Button!',
      style: 'secondary',
    });

    expect(result.html).toMatchSnapshot();
  });
});
