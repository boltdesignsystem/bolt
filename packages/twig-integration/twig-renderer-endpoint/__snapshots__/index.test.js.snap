// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Bolt Twig Renderer API renders attributes on the button component correctly 1`] = `
<bolt-button class="u-bolt-margin-bottom-large">
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center">
    <ssr-keep for="bolt-button"
              class="c-bolt-button__item"
    >
      Hello world!
    </ssr-keep>
  </button>
</bolt-button>
`;

exports[`Test the Bolt Twig Renderer API renders the button component correctly 1`] = `
<bolt-button>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center">
    <ssr-keep for="bolt-button"
              class="c-bolt-button__item"
    >
      Hello world!2
    </ssr-keep>
  </button>
</bolt-button>
`;

exports[`Test the Bolt Twig Renderer API renders the secondary button correctly 1`] = `
<bolt-button hierarchy="secondary">
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--secondary c-bolt-button--border-radius-regular c-bolt-button--center">
    <ssr-keep for="bolt-button"
              class="c-bolt-button__item"
    >
      Secondary Button!
    </ssr-keep>
  </button>
</bolt-button>
`;
